torch/compiler/__init__.py

```diff
@@ -0,0 +1,319 @@
+import torch
+import torch._dynamo
+import torch._inductor
+from typing import Callable, Union, List, Set, Tuple, Any, Dict
+from torch._dynamo.eval_frame import OptimizedModule
+
+__all__ = [
+    "OptimizedModule",
+    "compile",
+    "is_enabled",
+    "reset",
+    "allow_in_graph",
+    "list_backends",
+    "explain",
+    "is_compiling",
+    "disable",
+    "list_options",
+    "list_mode_options",
+]
+
+def compile(*args, **kwargs):
+    """
+    See :func:`torch.compile` for details on the arguments for this function.
+    """
+
+    return torch.compile(*args, **kwargs)
+
+def is_enabled() -> bool:
+    """
+    Check if compilation is enabled in the current environment.
+
+    This function checks if the `torch.compile` attribute exists in the `torch` module,
+    indicating that compilation is enabled. It returns `True` if compilation is enabled,
+    and `False` otherwise.
+
+    Returns:
+    - A boolean value indicating whether compilation is enabled (`True`) or not (`False`).
+
+    Note:
+    - The function checks the presence of the `torch.compile` attribute using the `hasattr()` function.
+    - If `torch.compile` exists, it indicates that compilation is enabled, and the function returns `True`.
+    - If `torch.compile` does not exist, it indicates that compilation is not enabled, and the function returns `False`.
+
+    """
+
+    if hasattr(torch, "compile"):
+        return True
+    else:
+        return False
+
+def reset() -> None:
+    """
+    Clear all compile caches and restore the initial state.
+
+    This function clears all compile caches and resets various internal state variables to their initial values.
+
+    It performs the following operations:
+    1. Clears the input and output code caches.
+    2. Clears the original code map.
+    3. Clears the guard failures.
+    4. Clears the cache for the `ContinueExecutionCache` class.
+    5. If the most recent backend in the evaluation frame has a 'reset' method, it is called.
+    6. Resets the most recent backend in the evaluation frame to None.
+    7. Clears the compilation metrics.
+    8. Resets the frame count to its initial value.
```

Comment:
These seem like implementation details, perhaps write in a more user-centric way.